src/SharpGLTF.Cesium/BinaryTable.cs

```diff
@@ -0,0 +1,208 @@
+            {
+                var vec = (Vector2)(object)values[i];
+                Buffer.BlockCopy(BitConverter.GetBytes(vec.X), 0, result, i * 8, 4);
+                Buffer.BlockCopy(BitConverter.GetBytes(vec.Y), 0, result, i * 8 + 4, 4);
+            }
+            return result;
+        }
+
+        private static byte[] Vector3ToBytes<T>(IReadOnlyList<T> values)
+        {
+            var result = new byte[values.Count * 12];
+            for (int i = 0; i < values.Count; i++)
+            {
+                var vec = (Vector3)(object)values[i];
+                Buffer.BlockCopy(BitConverter.GetBytes(vec.X), 0, result, i * 12, 4);
+                Buffer.BlockCopy(BitConverter.GetBytes(vec.Y), 0, result, i * 12 + 4, 4);
+                Buffer.BlockCopy(BitConverter.GetBytes(vec.Z), 0, result, i * 12 + 8, 4);
+            }
+            return result;
+        }
+
+        private static byte[] Vector4ToBytes<T>(IReadOnlyList<T> values)
+        {
+            var result = new byte[values.Count * 16];
+            for (int i = 0; i < values.Count; i++)
+            {
+                var vec = (Vector4)(object)values[i];
+                Buffer.BlockCopy(BitConverter.GetBytes(vec.X), 0, result, i * 16, 4);
+                Buffer.BlockCopy(BitConverter.GetBytes(vec.Y), 0, result, i * 16 + 4, 4);
+                Buffer.BlockCopy(BitConverter.GetBytes(vec.Z), 0, result, i * 16 + 8, 4);
+                Buffer.BlockCopy(BitConverter.GetBytes(vec.W), 0, result, i * 16 + 12, 4);
+            }
+            return result;
+        }
+
+        public static List<int> GetStringOffsets(List<string> values)
+        {
+            var offsets = new List<int>() { 0 };
+            foreach (var value in values)
+            {
+                var length = Encoding.UTF8.GetBytes(value).Length;
+                offsets.Add(offsets.Last() + length);
+            }
+
+            return offsets;
+        }
+
+
+        public static List<int> GetStringOffsets(List<List<string>> values)
+        {
+            var offsets = new List<int>() {};
+            foreach (var arr in values)
+            {
+                var arrOffsets = GetStringOffsets(arr);
+                var last = offsets.LastOrDefault();
+                foreach (var offset in arrOffsets)
+                {
+                    if(!offsets.Contains(last + offset))
+                    {
+                        offsets.Add(last + offset);
+                    }
+                }
+            }
+
+            return offsets;
+        }
+
+
+        public static List<int> GetArrayOffsets<T>(List<List<T>> values)
+        {
+            var offsets = new List<int>() { 0 };
+            foreach (var value in values)
+            {
+                offsets.Add(offsets.Last() + value.Count);
+
+            }
+            return offsets;
+        }
+
+        public static int GetSize<T>()
+        {
+            var isValueType = typeof(T).IsValueType;
```

Comment:
Use
```c#
var isValueType = !System.Runtime.CompilerServices.RuntimeHelpers.IsReferenceOrContainsReferences<T>();
```

Reply:
I'm getting a compilation error when using this RuntimeHelpers.IsReferenceOrContainsReferences: 

```
Error	CS0117	'RuntimeHelpers' does not contain a definition for 'IsReferenceOrContainsReferences'	SharpGLTF.Cesium (netstandard2.0)	
```

Without netstandard2.0 target framework there is no error (netstandard2.1;net6.0;net8.) remaining. Shall I remove netstandard2.0 from SharpGLTF.Cesium?

Reply:
you can do this:  #if NETSTANDARD  do this  #else  do that #endif

The advantage of IsReferenceOrContainsReferences is that it is more specific for blittable types.

Reply:
Fixed in https://github.com/vpenades/SharpGLTF/pull/206/commits/6dfc9a257e614530686b1525ec0331c4dca025d0

Reply:
I would reverse the condition to check explicitly for NETSTANDARD2.0 to use the .IsValue and everything else to use IsReferenceXXX

NETSTANDARD_X_OR_GREATER is deceiving because it makes you believe that it also includes net core, net 6 and older.... _it is not_  , it just applies to all Netstandard versions.

So if you want something that applies to NETSTANDARD2.1+ and Net6+  you have to explicitly state it, or invert the condition

Reply:
ok done in https://github.com/vpenades/SharpGLTF/pull/206/commits/e75dd9d40b55401462a9c2f5562d8d2d7df5824a